### PR TITLE
Add production deployment scripts and configuration files

### LINKED ISSUE #7: Set up deployment on production server
## Goal
Begin deploying application to production server with nginx

## Useful links

https://www.digitalocean.com/community/tutorials/how-to-set-up-a-node-js-application-for-production-on-ubuntu-20-04

https://www.digitalocean.com/community/tutorials/how-to-deploy-a-react-application-with-nginx-on-ubuntu-20-04

https://pm2.keymetrics.io/docs/tutorials/pm2-nginx-production-setup

https://gist.github.com/osharim/8263714

https://weia.xyz/tools/pm2/

## Acceptance criteria
- [x] Add documentation on deployment pipeline to Readme
- [x] Successfully serve site over the LANL network
- [x] Deploy Nginx + PM2 + Supervisord

### PR DESCRIPTION
**Pull request by dougUCN**
## Changes
- Adds instructions to `CONTRIBUTING.md` for deployment to production
- Adds a copy of the Nginx configuration file used in production to the repository
- Adds a copy of the PM2 `ecosystem.config.js` file used in production to the repository
- Adds a script for deployment to production (will need to be run on the production server manually)
Closes #7

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -98,3 +98,84 @@ The steps for setting up the client on AWS EC2 are outlined as follows:
...
+To update the production LANE-client, perform the following:
+
+1. Update the server on production as per the instructions [here](https://github.com/dougUCN/LANE-server)
+2. Run the `deploy.sh` script located in `$HOME/LANE/LANE-client/deploy-prod`
+
+```bash
+# Assuming you are in the $HOME/LANE/LANE-client directory
+chmod u+x deploy-prod/deploy.sh
+./deploy-prod/deploy.sh # Doesn't matter which directory from which you call deploy.sh
+```
```
Comment (khyeh) [inline CONTRIBUTING.md:115]: I think this comment can be removed.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -98,3 +98,84 @@ The steps for setting up the client on AWS EC2 are outlined as follows:
...
+10. If there is a file `/nginx/sites-enabled/default`, either delete the file or move it to the folder `/nginx/sites-available`
+11. If ufw is enabled, modify firewall rules with the command `sudo ufw allow 'Nginx Full'` (ONLY if ufw is enabled. By default on the production server it was not.)
+12. Restart nginx with `sudo systemctl restart nginx`
+13. Run the deploy shell script
+
+```bash
+# Assuming you are in the LANE-client directory
+chmod u+x deploy-prod/deploy.sh
+./deploy-prod/deploy.sh # Doesn't matter which directory from which you call deploy.sh
+```
```
Comment (khyeh) [inline CONTRIBUTING.md:177]: Same here

**Review by khyeh — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -98,3 +98,84 @@ The steps for setting up the client on AWS EC2 are outlined as follows:
...
+
+```
+sudo apt update
+sudo apt install nginx
+systemctl status nginx # Check nginx status
+```
+
+9.  Copy the contents of the file `/deploy-prod/lane.conf` to the nginx config file at `etc/nginx/conf.d`.
+10. If there is a file `/nginx/sites-enabled/default`, either delete the file or move it to the folder `/nginx/sites-available`
+11. If ufw is enabled, modify firewall rules with the command `sudo ufw allow 'Nginx Full'` (ONLY if ufw is enabled. By default on the production server it was not.)
```
Comment (khyeh) [inline CONTRIBUTING.md]: How can you tell whether or not ufw is enabled?

File: deploy-prod/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+echo "Jump to app folder"
+cd $HOME/LANE/LANE-client
+pwd
+
+git pull
+
+echo "Reinstalling dependencies"
+rm -rf node_modules
+npm ci
```
Comment (khyeh) [inline deploy-prod/deploy.sh:14]: I think `npm ci` automatically removes the existing `node_modules` directory. L.13 may be necessary.

File: deploy-prod/lane.conf [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# This is the nginx configuration file located at `/etc/nginx/conf.d/lane.conf`
+
+upstream my_nodejs_upstream {
+        server 127.0.0.1:3000;
+        keepalive 64;
+}
+
+server {
+    listen 80;
+            # listen 443 ssl;
```
Comment (khyeh) [inline deploy-prod/lane.conf]: What does this comment mean?

File: deploy-prod/deploy.sh (hunk 4 above)
Comment (dougUCN) [reply deploy-prod/deploy.sh:14]: Ah I see, yes I have confirmed that this is the case. Removed the `rm -rf`

File: CONTRIBUTING.md (hunk 1 above)
Comment (dougUCN) [reply CONTRIBUTING.md:115]: Removed

File: CONTRIBUTING.md (hunk 2 above)
Comment (dougUCN) [reply CONTRIBUTING.md:177]: Removed

File: deploy-prod/lane.conf (hunk 5 above)
Comment (dougUCN) [reply deploy-prod/lane.conf]: These are for when https is set up, which requires further configuration. Should I delete these or add some additional comment in this file?

File: CONTRIBUTING.md (hunk 3 above)
Comment (dougUCN) [reply CONTRIBUTING.md]: The command to check UFW status has been added

File: deploy-prod/lane.conf (hunk 5 above)
Comment (dougUCN) [reply deploy-prod/lane.conf]: Added a comment for clarification

**Pull request merged**